Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -758,37 +758,42 @@ private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
     return CodeBlockItemListSyntax(newItems)
   }
 
-  override func visit(_ node: MemberBlockItemListSyntax) -> MemberBlockItemListSyntax {
+  override func visit(_ node: MemberBlockSyntax) -> MemberBlockSyntax {
     let parentDeclGroup = node
-      .parent?
-      .as(MemberBlockSyntax.self)?
       .parent?
       .as(DeclSyntax.self)
     var newItems: [MemberBlockItemSyntax] = []
 
-    func addResult(_ node: MemberBlockItemSyntax) {
+    func addResult(_ memberBlockItem: MemberBlockItemSyntax) {
       // Expand freestanding macro.
-      switch expandMemberDecl(node: node) {
+      switch expandMemberDecl(node: memberBlockItem) {
       case .success(let expanded):
         for item in expanded {
           addResult(item)
         }
         return
       case .failure:
-        newItems.append(node)
+        newItems.append(memberBlockItem)
       case .notAMacro:
         // Recurse on the child node.
-        newItems.append(visit(node))
+
+        // Ensure there is only one newline if there is no member.
+        if newItems.isEmpty && node.members.isEmpty {
+          let singleNewlineTrivia = memberBlockItem.leadingTrivia.trimmingPrefix(while: { $0.isNewline })
+          newItems.append(visit(memberBlockItem.with(\.leadingTrivia, .newline + singleNewlineTrivia)))
+        } else {
+          newItems.append(visit(memberBlockItem))
+        }
```

Comment:
I'm not sure if this is the right approach. 

In MacroSystem.swift:212 we add `\n\n`. And here I want to override it

Reply:
@bnbarham do you also think it's fine? 
You mentioned here that you would expect without: https://github.com/apple/swift-syntax/pull/2320#discussion_r1406497566

Reply:
I don't really mind. I think it would be nice to remove and this isn't that much of a hack, but on the other hand the two `\n\n` is what the compiler sees, so you could argue it makes sense to keep them. As long as we add a newline before the `}`, I'm happy :)